schema_based_on_diagram.sql

```diff
@@ -0,0 +1,48 @@
+-- Create patients table
+CREATE TABLE patients (
+  id INT PRIMARY KEY,
+  name VARCHAR(255) NOT NULL,
+  date_of_birth DATE NOT NULL
+);
+
+-- Create medical_histories table
+CREATE TABLE medical_histories (
+  id INT PRIMARY KEY,
+  admitted_at timestamp NOT NULL,
+  patient_id INT NOT NULL,
+  treatments_id INT NOT NULL,
+  status VARCHAR(255) NOT NULL,
+  FOREIGN KEY (patient_id) REFERENCES patients(id)
+  FOREIGN KEY (treatments_id) REFERENCES treatments(id)
+);
+
+-- Create incoices table
+CREATE TABLE invoices (
+  id INT PRIMARY KEY,
+  total_amount DECIMAL(10,2) NOT NULL,
+  generated_at timestamp,
+  payed_at timestamp,
+  medical_history_id INT NOT NULL,
+  FOREIGN KEY (medical_history_id) REFERENCES medical_histories(id)
+);
+
+-- Create treatments table
+CREATE TABLE treatments (
+  id INT PRIMARY KEY,
+  name VARCHAR(255) NOT NULL,
+  type VARCHAR(255) NOT NULL,
+  medical_history_id INT NOT NULL,
+  FOREIGN KEY (medical_history_id) REFERENCES medical_histories(id)
+);
+
+-- Create invoice_items table
+CREATE TABLE invoice_items (
+  id INT PRIMARY KEY,
+  unit_price DECIMAL(10,2) NOT NULL,
+  quantity INT NOT NULL,
+  total_price DECIMAL(10,2) NOT NULL,
+  invoice_id INT NOT NULL,
+  treatment_id INT NOT NULL,
+  FOREIGN KEY (invoice_id) REFERENCES invoices(id),
+  FOREIGN KEY (treatment_id) REFERENCES treatments(id)
+);
```

Comment:
- [ ] As it is  a [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/database_from_diagram.md) for this project: "`Join tables from many-to-many relationships might not appear in the diagram, but you still need them`", There is a many-to-many relationship between `medical_histories` and `treatments`, please kindly create a join table for that. Something like so (just an example): 
````sql
  CREATE TABLE medical_histories_and_treatments (
    medical_history_id int refrences medical_histories(id),
    treatment_id int refrences treatments(id),
    );
```

Reply:
Thanks. we forgot that a many-to-many relationship required a separate table